src/blocks/scratch3_operators.js

```diff
@@ -31,8 +40,54 @@ Scratch3OperatorsBlocks.prototype.add = function (args) {
     return args.NUM1 + args.NUM2;
 };
 
+Scratch3OperatorsBlocks.prototype.subtract = function (args) {
+    return args.NUM1 - args.NUM2;
+};
+
+Scratch3OperatorsBlocks.prototype.multiply = function (args) {
+    return args.NUM1 * args.NUM2;
+};
+
+Scratch3OperatorsBlocks.prototype.divide = function (args) {
+    return args.NUM1 / args.NUM2;
+};
+
+Scratch3OperatorsBlocks.prototype.lt = function (args) {
+    return args.OPERAND1 < args.OPERAND2;
+};
+
 Scratch3OperatorsBlocks.prototype.equals = function (args) {
     return args.OPERAND1 == args.OPERAND2;
 };
 
+Scratch3OperatorsBlocks.prototype.gt = function (args) {
+    return args.OPERAND1 > args.OPERAND2;
+};
+
+Scratch3OperatorsBlocks.prototype.and = function (args) {
+    if (!args.OPERAND1 || !args.OPERAND2) {
+        return false;
+    }
+    return true;
+};
+
+Scratch3OperatorsBlocks.prototype.or = function (args) {
+    return args.OPERAND1 || args.OPERAND2;
+};
+
+Scratch3OperatorsBlocks.prototype.not = function (args) {
+    return !args.OPERAND;
+};
+
+Scratch3OperatorsBlocks.prototype.random = function (args, util) {
+    // As a demo, this implementation of random returns after 1 second of yield.
+    // @todo Match Scratch 2.0 implementation with int-truncation.
+    // See: http://bit.ly/1Qc0GzC
+    util.yieldAndBlock();
+    setTimeout(function() {
+        var randomValue = (Math.random() * (args.TO - args.FROM)) + args.FROM;
+        util.report(randomValue);
```

Comment:
@cwillisf Check out my latest commit - updated it as you suggested. I think right now it doesn't make the implementation much simpler, but I still like it, especially how well it would interact with the renderer, things like https://github.com/request/request-promise, etc.